packages/crypto/src/crypto-primitives/pbkdf2.ts

```diff
@@ -0,0 +1,78 @@
+import { crypto } from '@noble/hashes/crypto';
+
+import { isWebCryptoSupported } from '../utils.js';
+
+type DeriveKeyOptions = {
+  hash: 'SHA-256' | 'SHA-384' | 'SHA-512',
+  password: Uint8Array,
+  salt: Uint8Array,
+  iterations: number,
+  length: number
+};
+
+export class Pbkdf2 {
+  public static async deriveKey(options: DeriveKeyOptions): Promise<Uint8Array> {
+    if (isWebCryptoSupported()) {
+      return Pbkdf2.deriveKeyWithWebCrypto(options);
+    } else {
+      return Pbkdf2.deriveKeyWithNodeCrypto(options);
+    }
+  }
+
+  private static async deriveKeyWithNodeCrypto(options: DeriveKeyOptions): Promise<Uint8Array> {
+    const { password, salt, iterations } = options;
+
+    // Map the hash string to the node:crypto equivalent.
+    const hashToNodeCryptoMap = {
+      'SHA-256' : 'sha256',
+      'SHA-384' : 'sha384',
+      'SHA-512' : 'sha512'
+    };
+    const hash = hashToNodeCryptoMap[options.hash];
+
+    // Convert length from bits to bytes.
+    const length = options.length / 8;
+
+    // Dynamically import the `crypto` package.
+    const { pbkdf2 } = await import('node:crypto');
```

Comment:
@shamilovtim I tried swapping this for just `crypto` but it causes the browser bundle build to fail:

```
> rimraf dist/browser.mjs dist/browser.js && node build/bundles.js

✘ [ERROR] Could not resolve "crypto"

    src/crypto-primitives/pbkdf2.ts:37:36:
      37 │     const { pbkdf2 } = await import('crypto');
         ╵                                     ~~~~~~~~

  The package "crypto" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
  ```

Any tricks you can think of to still polyfill with `react-native-quick-crypto` for RN?

Reply:
Let me try sending in a PR for this